src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt

```diff
@@ -273,26 +273,30 @@ abstract class FixupSession(
   fun performWorkspaceEdit(workspaceEditParams: WorkspaceEditParams) {
     for (op in workspaceEditParams.operations) {
       // TODO: We need to support the file-level operations.
-      when (op.type) {
-        "create-file" -> {
-          logger.warn("Workspace edit operation created a file: ${op.uri}")
-        }
-        "rename-file" -> {
-          logger.warn("Workspace edit operation renamed a file: ${op.oldUri} -> ${op.newUri}")
-        }
-        "delete-file" -> {
-          logger.warn("Workspace edit operation deleted a file: ${op.uri}")
-        }
-        "edit-file" -> {
-          if (op.edits == null) {
-            logger.warn("Workspace edit operation has no edits")
-          } else {
-            performInlineEdits(op.edits)
+      try {
+        when (op.type) {
+          "create-file" -> {
+            logger.warn("Workspace edit operation created a file: ${op.uri}")
+          }
+          "rename-file" -> {
+            logger.warn("Workspace edit operation renamed a file: ${op.oldUri} -> ${op.newUri}")
+          }
+          "delete-file" -> {
+            logger.warn("Workspace edit operation deleted a file: ${op.uri}")
+          }
+          "edit-file" -> {
+            if (op.edits == null) {
+              logger.warn("Workspace edit operation has no edits")
+            } else {
+              performInlineEdits(op.edits)
+            }
           }
+          else ->
+              logger.warn(
+                  "DocumentCommand session received unknown workspace edit operation: ${op.type}")
         }
-        else ->
-            logger.warn(
-                "DocumentCommand session received unknown workspace edit operation: ${op.type}")
+      } catch (e: RuntimeException) {
+        throw WorkspaceEditException(op, e)
```

Comment:
If we hope it will help us debug e.g. issue with indexes, it would be much more useful to do those try/caches at the level of single edit, and that is what is causing an issue.

Than you also have much more rich and useful info to log:

```kt
data class TextEdit(
    // This tag will be 'replace', 'insert', or 'delete'.
    val type: String,

    // Valid for replace & delete.
    val range: Range? = null,

    // Valid for insert.
    val position: Position? = null,

    // Valid for replace & insert.
    val value: String? = null
)
```

I think we should also log the data of the opened document: its url and its length.

Reply:
Refactored.